weatherAplication/Modules/Search/View/SearchViewController.swift

```diff
@@ -67,18 +67,36 @@ class SearchViewController: UIViewController {
 
 extension SearchViewController: UITableViewDelegate, UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return cityInfosArray.count
+        return viewModel.cityInfoArray.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath as IndexPath)
-        cell.textLabel?.text = cityInfosArray[indexPath.row].city
+        cell.textLabel?.text = viewModel.cityInfoArray[indexPath.row].city
         return cell
     }
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let cityInfoToSend = cityInfosArray[indexPath.row]
-        let nextViewController = ShowWeatherViewController(data: cityInfoToSend)
+        let cityInfoToSend = viewModel.cityInfoArray[indexPath.row]
+        cityChosen(cityInfo: cityInfoToSend)
+    }
+}
+
+extension SearchViewController: SearchViewModelDelegate {
+    func cityChosen(cityInfo: CityInfo) {
+        let nextViewController = ShowWeatherViewController(data: cityInfo)
```

Comment:
tutaj lepiej by bylo zrobic ShowWeatherViewController(viewModel: ShowWeatherViewModel(data: cityInfo)), ale tak naprawde w przypadku takiej malej aplikacji, w ktorej nie jest stosowane dependency injection to nie ma tak duzego znaczenia, wiec up to you czy chcesz to poprawic

Reply:
Wydaje mi się że jednak nie, taka technika jest spoko ale tylko jak się używa coordynatorów, a Adam jednak na razie tego nie ma